cmd/flux/cmd/test.go

```diff
@@ -46,7 +46,9 @@ func TestCommand(setup TestSetupFunc) *cobra.Command {
 		Long:  "Run flux tests",
 		Run: func(cmd *cobra.Command, args []string) {
 			fluxinit.FluxInit()
-			runFluxTests(setup, flags)
+			if err := runFluxTests(setup, flags); err != nil {
+				panic(err)
```

Comment:
You can call `os.Exit()` here safely. That is more correct than using a panic.

Reply:
Sure, I can try that. I ran into an edge case on the idpe side that prompted me to push the deferred cleanup closer to the top of the call stack, where an Exit here would interfere (causing a forked process to outlive the test harness).

I'll see if I can get around the issue for idpe and try  this out.

Reply:
The gist of the issue is around this:

```
	executor, err := setup(context.Background())
	if err != nil {
		return err
	}
	defer func() { _ = executor.Close() }() // never called if setup gives an err

	runner.Run(executor, flags.verbosity, flags.skipTestCases)
	return runner.Finish()
}
```

For the idpe w/ IOx use case, setup will try to fork a process to run IOx mid-way through, with later parts of the impl possibly giving errors.

To ensure the forked process gets killed on the way out, we therefore need something higher in the call stack.

This `Run` callback is the callsite for this function, so if it calls `Exit`, the cleanup can't happen (since it would be waiting, deferred higher-up still).

For reference, this is how the `Command` is invoked:

```
func tryExec(cmd *cobra.Command) (err error) {
	defer func() {
		if e := recover(); e != nil {
			var ok bool
			err, ok = e.(error)
			if !ok {
				err = errors.New(fmt.Sprint(e))
			}
		}
	}()
	defer func() { _ = killIOx() }()
	err = cmd.Execute() // the Exit call would hit here, meaning the above defer will not fire
	return
}

func main() {
	c := cmd.TestCommand(NewTestExecutor)
	c.Use = "fluxtest"
	if err := tryExec(c); err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
}
```

Reply:
For completeness -- there was room to kill the forked process inside the setup function _after detecting an error in `init`_. This meant calling `Exit` in the command's callback is no longer an issue for us.